kiwi/bootloader/config/grub2.py

```diff
@@ -716,3 +719,24 @@ def _find_grub_data(self, lookup_path):
         raise KiwiBootLoaderGrubDataError(
             'No grub2 installation found in %s' % lookup_path
         )
+
+    def _get_grub_boot_directory_name(self):
+        """
+        Get grub data directory name in boot/ directory
+
+        Depending on the distribution the grub boot path could be
+        either boot/grub2 or boot/grub. The method will decide for
+        the correct base directory name according to the name pattern
+        of the installed grub tools
+        """
+        chroot_env = {
+            'PATH': os.sep.join([self.root_dir, 'usr', 'sbin'])
+        }
+        if Path.which(filename='grub-install', custom_env=chroot_env):
```

Comment:
You'll want to test for grub2-* first, since some distros ship both grub legacy and grub2, which would confuse this test